tomopy/prep/alignment.py

```diff
@@ -443,3 +455,240 @@ def shift_images(prj, sx, sy):
+                                    [cnt[0], cnt[1]],  # lower right, 3rd quadrant
+                                    [cnt[1],      0],  # upper right, 4th quadrant
+                                   ])
+        # init four corners
+        cnrs = np.zeros((4, 2))
+        if method.lower() == 'quadrant':
+            # the standard quadrant method
+            for i, q in enumerate(quadrants):
+                cnrs[i,:] = np.array([q.col_func(np.gradient(medfilt(np.std(q.img, axis=0), kernel_size=medfilt_kernel_size))),  # x is col_idx
+                                      q.row_func(np.gradient(medfilt(np.std(q.img, axis=1), kernel_size=medfilt_kernel_size))),  # y is row_idx
+                                    ])
+            # add the origin offset back
+            cnrs = cnrs + quadrantOrigins
+        elif method.lower() == 'quadrant+':
+            # use Gaussian curve fitting to achive subpixel precision 
+            # TODO:
+            #   improve the curve fitting with Lorentz and Voigt fitting function
+            for i, q in enumerate(quadrants):
+                # -- find x subpixel position
+                cnr_x_guess = q.col_func(np.gradient(medfilt(np.std(q.img, axis=0), kernel_size=medfilt_kernel_size)))
+                # isolate the strongest peak to fit
+                tmpx = np.arange(cnr_x_guess-10, cnr_x_guess+11)
+                tmpy = np.gradient(np.std(q.img, axis=0))[tmpx]
+                # tmpy[0] is the value from the highest/lowest pixle
+                # tmpx[0] is basically cnr_x_guess
+                # 5.0 is the guessted std, 
+                coeff, _ = curve_fit(gauss, tmpx, tmpy, 
+                                     p0=[tmpy[0], tmpx[0], 5.0],
+                                     maxfev=int(1e6),
+                                    )
+                cnrs[i, 0] = coeff[1]  # x position
+                # -- find y subpixel positoin
+                cnr_y_guess = q.row_func(np.gradient(medfilt(np.std(q.img, axis=1), kernel_size=medfilt_kernel_size)))
+                # isolate the peak (x, y here is only associated with the peak)
+                tmpx = np.arange(cnr_y_guess-10, cnr_y_guess+11)
+                tmpy = np.gradient(np.std(q.img, axis=1))[tmpx]
+                coeff, _ = curve_fit(gauss, tmpx, tmpy, 
+                                     p0=[tmpy[0], tmpx[0], 5.0],
+                                     maxfev=int(1e6),
+                                    )
+                cnrs[i, 1] = coeff[1]  # y posiiton
+            # add the quadrant shift back 
+            cnrs = cnrs + quadrantOrigins
+                
+        else:
+            raise NotImplementedError("Available methods are: simple, quadrant, quadrant+")
+    
+    # return the slit corner detected
+    return cnrs
+
+
+def calc_slit_box_aps_1id(slit_box_corners, inclip=(1,10,1,10)):
+    """
+    Calculate the clip box based on given slip corners

```

Comment:
What is a clip box? What is a slit? Some definition of these is appreciated because it seems domain specific.

Reply:
ASCII art is already added in the docstring.